src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetCommandHandler.cs

```diff
@@ -240,6 +240,12 @@ protected bool TryHandleTypedSnippet(ITextView textView, ITextBuffer subjectBuff
         {
             AssertIsForeground();
 
+            // Don't execute in cloud environment, should be handled by LSP
+            if (subjectBuffer.IsInCloudEnvironmentClientContext())
```

Comment:
I think we should just move this check to AreSnippetsEnabled, that way it happens earlier and still covers all entry points

Reply:
Moved!